From 3f54d1911308b42e0063730087d48880514b9eeb Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 01 Feb 2022 19:01:00 +0200
Subject: [PATCH]  [SW-69188] common: move export function file device specific

Since new APIs will be exported started from gaudi3, this patch add
change to move export text file device specific.

Change-Id: I646fcf1e6230e6277f2685fefa4c6aa0166e46e5
---

diff --git a/Makefile b/Makefile
index b9d839a..0bef4b5 100644
--- a/Makefile
+++ b/Makefile
@@ -79,7 +79,7 @@
 	@$(CC) $^ $(LDFLAGS) -Wl,--just-symbols=exports.elf -o $@
 arcboot_rom: $(ARCBOOT_ROM_OBJECTS)
 	@$(CC) $^ $(LDFLAGS) -o $@
-	@$(OBJCOPY) --strip-all --keep-symbols=app/$@.txt $@ exports.elf
+	@$(OBJCOPY) --strip-all --keep-symbols=$(HL_DEV_ARCH)/$@.txt $@ exports.elf
 
 %.o: %.c
 	@$(CC) $(ARCBOOT_CFLAGS) -c $< -o $@
diff --git a/app/arcboot_rom.txt b/app/arcboot_rom.txt
deleted file mode 100644
index 92277c6..0000000
--- a/app/arcboot_rom.txt
+++ /dev/null
@@ -1,6 +0,0 @@
-# put list of symbols, one per line, to be exported
-dma_wait_finish
-security_i2c_s_enable
-secureboot_verify_image
-dma_copy_image
-spi_flash_readl
diff --git a/arch/gaudi2/arc/arcboot_rom.txt b/arch/gaudi2/arc/arcboot_rom.txt
new file mode 100644
index 0000000..92277c6
--- /dev/null
+++ b/arch/gaudi2/arc/arcboot_rom.txt
@@ -0,0 +1,6 @@
+# put list of symbols, one per line, to be exported
+dma_wait_finish
+security_i2c_s_enable
+secureboot_verify_image
+dma_copy_image
+spi_flash_readl
diff --git a/arch/gaudi3/arc/arcboot_rom.txt b/arch/gaudi3/arc/arcboot_rom.txt
new file mode 100644
index 0000000..92277c6
--- /dev/null
+++ b/arch/gaudi3/arc/arcboot_rom.txt
@@ -0,0 +1,6 @@
+# put list of symbols, one per line, to be exported
+dma_wait_finish
+security_i2c_s_enable
+secureboot_verify_image
+dma_copy_image
+spi_flash_readl
diff --git a/arch/greco/arc_elf32/arcboot_rom.txt b/arch/greco/arc_elf32/arcboot_rom.txt
new file mode 100644
index 0000000..92277c6
--- /dev/null
+++ b/arch/greco/arc_elf32/arcboot_rom.txt
@@ -0,0 +1,6 @@
+# put list of symbols, one per line, to be exported
+dma_wait_finish
+security_i2c_s_enable
+secureboot_verify_image
+dma_copy_image
+spi_flash_readl
